extensions/cache/runtime/src/main/java/io/quarkus/cache/CompositeCacheKey.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.cache;
+
+import java.util.Arrays;
+
+/**
+ * A composite cache key is used by the annotations caching API when a method annotated with {@link CacheResult} or
+ * {@link CacheInvalidate} is invoked and when the cache key is composed of several of the method arguments (annotated with
+ * {@link CacheKey} or not). This class can also be used with the programmatic caching API.
```

Comment:
What would be the use case for the programmatic API (I'd be tempted to have my own business centric composite object).
Is it to interact with the elements cached via annotation? We could call out that use case explicitly

Reply:
> Is it to interact with the elements cached via annotation?

Yes, that's the reason I moved this class to the API package in this PR. This was asked in #6903.

Reply:
The doc current doc should answer that question.